jdaviz/configs/default/plugins/model_fitting/fitting_backend.py

```diff
@@ -251,7 +255,13 @@ def __call__(self):
             # to execute. This behavior was seen also with other functions
             # passed to the callable.
             flux = self.cube[x, y, :]
-            sp = Spectrum1D(spectral_axis=self.wave, flux=flux)
+            if self.mask is not None:
+                mask = self.mask[x, y, :]
+            else:
+                # If no mask is provided:
+                mask = np.zeros_like(flux).astype(bool)
+
+            sp = Spectrum1D(spectral_axis=self.wave, flux=flux, mask=mask)
```

Comment:
What if `mask = self.mask[x, y, :]` is all `True`? Is that possible? In that case, what is the point running this? Should this abort early in that case?

Or should we only extract the unmasked flux here instead of passing in `mask` or is `specutils` smart enough to optimize resources if a big array is passed in but most of it is masked out?

Reply:
I suspect that no Cubeviz behavior when trying to fit a fully-masked spectral cube will prevent users from being disappointed.

Reply:
Not sure if I understand. Now that we support mask, new use cases should be tested. What if mask is very small, completely out of bounds, partial coverage, etc.

Reply:
jdaviz does the fit via the `specutils.fitting.fit_lines` function, which [uses `astropy.fitting.LevMarLSQFitter`](https://github.com/astropy/astropy/blob/b2f33b392c9d470d7f839e6135485aac6a1c5f4a/astropy/modeling/fitting.py#L1268) under the hood, which in turn [relies on `scipy.optimize.leastsq`](https://github.com/astropy/astropy/blob/b2f33b392c9d470d7f839e6135485aac6a1c5f4a/astropy/modeling/fitting.py#L1313-L1320). The [`leastsq` docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.leastsq.html) don't mention masking, and the [scipy source](https://github.com/scipy/scipy/blob/v1.9.3/scipy/optimize/_minpack_py.py#L279-L479) isn't illuminating either. To summarize: I don't know where the masking gets implemented.

Reply:
I think @pllim's point is most directed at spatial subsets - if the user is fitting a 10x10 square spatial subset, for example, can we speed this up but bailing early on the other 5000 spaxels rather than passing fully masked Spectrum1D objects to the lower level fitting code at all? It's the case of the whole spaxel being masked, not the whole cube being masked.

Reply:
> It's the case of the whole spaxel being masked, not the whole cube being masked.

Ah _this_ is what I didn't understand correctly in my first reads of @pllim's comment, sorry for my misinterpretation.